multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -0,0 +1,291 @@
+const { isGTE, isEmpty, make } = AmountMath;
+
+const test = anyTest as TestFn<
+  SetupContextWithWallets & {
+    lpUser: WalletDriver;
+    oracleWds: WalletDriver[];
+    nobleAgoricChannelId: IBCChannelID;
+    usdcOnOsmosis: Denom;
+  }
+>;
+
+const accounts = [...keys(oracleMnemonics), 'lp'];
+const contractName = 'fastUsdc';
+const contractBuilder =
+  '../packages/builders/scripts/fast-usdc/init-fast-usdc.js';
+
+test.before(async t => {
+  const { setupTestKeys, ...common } = await commonSetup(t);
+  const {
+    chainInfo,
+    commonBuilderOpts,
+    deleteTestKeys,
+    faucetTools,
+    provisionSmartWallet,
+    startContract,
+  } = common;
+  deleteTestKeys(accounts).catch();
+  const wallets = await setupTestKeys(accounts, values(oracleMnemonics));
+
+  // provision oracle wallets first so invitation deposits don't fail
+  const oracleWds = await Promise.all(
+    keys(oracleMnemonics).map(n =>
+      provisionSmartWallet(wallets[n], {
+        BLD: 100n,
+      }),
+    ),
+  );
+
+  // calculate denomHash and channelId for privateArgs / builder opts
+  const { getTransferChannelId, toDenomHash } = makeDenomTools(chainInfo);
+  const usdcDenom = toDenomHash('uusdc', 'noblelocal', 'agoric');
+  const usdcOnOsmosis = toDenomHash('uusdc', 'noblelocal', 'osmosis');
+  const nobleAgoricChannelId = getTransferChannelId('agoriclocal', 'noble');
+  if (!nobleAgoricChannelId) throw new Error('nobleAgoricChannelId not found');
+  t.log('nobleAgoricChannelId', nobleAgoricChannelId);
+  t.log('usdcDenom', usdcDenom);
+
+  await startContract(contractName, contractBuilder, {
+    oracle: keys(oracleMnemonics).map(n => `${n}:${wallets[n]}`),
+    usdcDenom: usdcDenom,
+    feedPolicy: makeFeedPolicy(nobleAgoricChannelId),
+    ...commonBuilderOpts,
+  });
+
+  // provide faucet funds for LPs
+  await faucetTools.fundFaucet([['noble', 'uusdc']]);
+
+  // save an LP in test context
+  const lpUser = await provisionSmartWallet(wallets['lp'], {
+    USDC: 100n,
+    BLD: 100n,
+  });
+
+  t.context = {
+    ...common,
+    lpUser,
+    oracleWds,
+    nobleAgoricChannelId,
+    usdcOnOsmosis,
+    wallets,
+  };
+});
+
+test.after(async t => {
+  const { deleteTestKeys } = t.context;
+  deleteTestKeys(accounts);
+});
+
+const toOracleOfferId = (idx: number) => `oracle${idx + 1}-accept`;
+
+test.serial('oracles accept', async t => {
+  const { oracleWds, retryUntilCondition, vstorageClient, wallets } = t.context;
+
+  const instances = await vstorageClient.queryData(
+    'published.agoricNames.instance',
+  );
+  const instance = fromEntries(instances)[contractName];
+
+  // accept oracle operator invitations
+  await Promise.all(
+    oracleWds.map(makeDoOffer).map((doOffer, i) =>
+      doOffer({
+        id: toOracleOfferId(i),
+        invitationSpec: {
+          source: 'purse',
+          instance,
+          description: 'oracle operator invitation', // TODO export/import INVITATION_MAKERS_DESC
```

Comment:
yeah; we should have a helper in `@agoric/fast-usdc/tools/` for building this whole offer spec.

Reply:
check out https://github.com/Agoric/agoric-sdk/blob/master/packages/inter-protocol/src/clientSupport.js#L429-L455